### ISSUE TITLE
"Make readonly" doesn't appear for some simple cases

### ISSUE THREAD

**Issue by kuhlenh** [dotnet/roslyn#26213]
'Make readonly’ doesn't seem to work for a really simple case:

```csharp
public static class Function1
    {

        private static HttpClient client = new HttpClient();

        private static IFaceServiceClient faceServiceClient = new FaceServiceClient("key", "https://westus.api.cognitive.microsoft.com/face/v1.0");

        public static IEnumerable<FaceAttributeType> faceAttributes = new FaceAttributeType[] { FaceAttributeType.Gender, FaceAttributeType.Age, FaceAttributeType.Smile, FaceAttributeType.Emotion, FaceAttributeType.Glasses, FaceAttributeType.Hair };

        [FunctionName("Function1")]
        public static async Task Run([BlobTrigger("samples-workitems/{name}", Connection = "")]Stream myBlob, string name, TraceWriter log)
        {
            try
            {
                Face[] faces = await faceServiceClient.DetectAsync(myBlob, returnFaceId: true, returnFaceLandmarks: false, returnFaceAttributes: faceAttributes);

                foreach (Face face in faces)
                {
                    var emotionScores = face.FaceAttributes.Emotion.Happiness;
                }
            }
            catch (FaceAPIException f)
            {
                log.Info($"{f.ErrorMessage}, {f.ErrorCode}");
            }
            catch (Exception e)
            {
                log.Info($"{e.Message}");
            }
        }
    }
```

- “Make readonly” appears for the HttpClient
- “Make readonly” **does not appear** for IFaceServiceClient

**Comment by CyrusNajmabadi** [comment]
I see it:

![image](https://user-images.githubusercontent.com/4564579/38900296-d1420c3c-424f-11e8-98ba-a443b628b23c.png)

**Comment by sharwell** [comment]
@kuhlenh It seems the code snippet to reproduce this is incomplete.

**Comment by kuhlenh** [comment]
Figured it out. It's the try-catch. Updated the original repro

**Comment by CyrusNajmabadi** [comment]
Much simpler repro:

```c#
interface IFaceServiceClient
{
    void DetectAsync();
}

public class Repro
{
    private static IFaceServiceClient faceServiceClient = null;

    public static void Run()
    {
        faceServiceClient.DetectAsync();
    }
}
```

**Comment by CyrusNajmabadi** [comment]
Most of the code here seems completely wrong.  There's also this check:

```c#
            // make sure it isn't in an object initializer
            if (node.Parent.Parent is InitializerExpressionSyntax)
            {
                return false;
            }
```

This seems like it may also be wrong if you're on the initializer, and not the name, of an object initializer.

**Comment by Neme12** [comment]
I would just fix this as part of #25652

**Comment by sharwell** [comment]
@Neme12 There are some major hurdles (performance related) that could stall #25652. If this is an easy fix it could certainly be relevant in the 15.8 time frame.

**Comment by CyrusNajmabadi** [comment]
@sharwell What are the major hurdles?  Is it IOperation related?  I used IOperation in several recent PRs and I'm curious if htat will be a problem.
